src/mrinufft/operators/interfaces/cufinufft.py

```diff
@@ -850,3 +850,55 @@ def toggle_grad_traj(self):
         if self.uses_sense:
             self.smaps = self.smaps.conj()
         self.raw_op.toggle_grad_traj()
+
+    @classmethod
+    def pipe(
+        cls,
+        kspace_loc,
+        volume_shape,
+        num_iterations=10,
+        osf=2,
+        normalize=True,
+        **kwargs,
+    ):
+        """Compute the density compensation weights for a given set of kspace locations.
+
+        Parameters
+        ----------
+        kspace_loc: np.ndarray
+            the kspace locations
+        volume_shape: np.ndarray
+            the volume shape
+        num_iterations: int default 10
+            the number of iterations for density estimation
+        osf: float or int
+            The oversampling factor the volume shape
+        normalize: bool
+            Whether to normalize the density compensation.
+            We normalize such that the energy of PSF = 1
+        """
+        if CUFINUFFT_AVAILABLE is False:
+            raise ValueError(
+                "cufinufft is not available, cannot estimate the density compensation"
+            )
+        grid_op = MRICufiNUFFT(
```

Comment:
Since this is a classmethod, use `cls` instead of the backend name. Ultimately this could be refactorized between all backend that supports "pipe" density (with an extra attention to the cupy/numpy array interface)